README.md

```diff
@@ -15,3 +15,22 @@ the function and run automatically when you commit through a GitHub action.
 
 What is the worst-case big $\Theta$ complexity of your implementation? Add your
 answer, including your reasoning, to this markdown file.
+
+The worst case time complexity for the implemented code is $\Theta(V + E)$. In this V represents the vertices and E represents the edges. This is because the depth first search starts iterating over any vertices that are not visited which takes time complexity of $O(V)$. In each iteration it traverses through evry edge of that vertice to find the cycle which takes the time complexity of $O(E)$. So in total we end up with worst case time complexity of $\Theta(V + E)$.
+
+References:
+
+https://www.geeksforgeeks.org/detect-cycle-undirected-graph/
+
+The above website gave a straight forward understanding of the detecting cycles
+
+https://www.geeksforgeeks.org/depth-first-search-or-dfs-for-a-graph/
+
+The above website gave me a good idea of doing DFS
+
+
+I looked at the following repository for understanding how they initialized their visited array. However I made sure that I set everything inside the visited array to be false. So I can mark one by one as the nodes are visited.
```

Comment:
Which repository?

Reply:
Sorry about that. I missed to add the link